functions/src/investigationCall.ts

```diff
@@ -36,13 +36,15 @@ export const onMessageCreate = functions.firestore
             
             if(res.status !== 200){
                 console.error(res.data);
-                throw new functions.https.HttpsError('internal', 'Failed to call OpenAI API');
+                throw new functions.https.HttpsError('internal', 'Failed to call OpenAI API', 
+                    { userMessage: 'There was a problem with the AI assistant.Please try again later.'});
             }
 
             const message = res.data.choices[0]?.message;
             if(!message){
                 console.error('No answer from assistant');
-                throw new functions.https.HttpsError('not-found', 'No answer from assistant' )
+                throw new functions.https.HttpsError('not-found', 'No answer from assistant',
+                    { userMessage: 'The AI assistant did not provide a response. Please try again.' } )
             }
 
             await messagesRef.add({
```

Comment:
In case of error this line will never get reached. Therefore, the user will not understand what happened. Let's save the error to this document by adding another field called 'error' if it occurs so we could show it on the screen. 

if there is no error, 'error' field should not be added to the document.